devito/__init__.py

```diff
@@ -23,25 +25,40 @@
 __version__ = get_versions()['version']
 del get_versions
 
-# First add the compiler configuration option...
 configuration.add('compiler', 'custom', list(compiler_registry),
                   callback=lambda i: compiler_registry[i]())
+configuration.add('backend', 'core', list(backends_registry), callback=init_backend)
 
-
-def _cast_and_update_compiler(val):
+# OpenMP setup
+def _cast_and_update_compiler(val):  # noqa
     # Force re-build the compiler
     configuration['compiler'].__init__(suffix=configuration['compiler'].suffix)
     return bool(val)
+configuration.add('openmp', 0, [0, 1], callback=_cast_and_update_compiler)  # noqa
+
+# Autotuning setup
+AT_LEVELs = ['off', 'basic', 'aggressive']
+AT_MODEs = ['preemptive', 'runtime']
+at_default_mode = {'core': 'preemptive', 'yask': 'runtime'}
+at_setup = namedtuple('at_setup', 'level mode')
+at_accepted = AT_LEVELs + [list(i) for i in product(AT_LEVELs, AT_MODEs)]
+def _at_callback(val):  # noqa
+    if isinstance(val, str):
+        level, mode = val, at_default_mode[configuration['backend']]
+    else:
```

Comment:
Just a random thought - are the callbacks evaluated in a deterministic order?  Is `configuration['backend']` necessarily set to its proper value at this point?  What if it is changed?

Reply:
uops, I had forgotten about this. 

Is this what you mean: https://github.com/opesci/devito/pull/601/files#diff-2d5ca91610163c3bffccbb775c230dc1R124 
?